src/PhpDoc/PhpDocTypeHelper.php

```diff
@@ -194,4 +213,53 @@ private static function handleIdentifierNode(IdentifierTypeNode $type)
 
         return new ObjectType($type->name);
     }
+
+    /**
+     * @param  list<\PHPStan\PhpDocParser\Ast\Type\TypeNode>  $genericTypes
+     */
+    private static function handleGenericInteger(array $genericTypes): IntegerType
```

Comment:
I personally don't like the amount of if statements that are in this method, might look into it later